### ISSUE TITLE
Admin UI breaks after inserting invalid relation

### ISSUE THREAD

**Issue by bictorman** [pocketbase/pocketbase#3372]
**Version**: 0.18.6
**Platform**: Mac (probably all)
**Description**: Inserting a relationship with an invalid ID breaks the admin UI for that table completely, no data is displayed and the table can not be edited or deleted.
<img width="1090" alt="Screenshot 2023-09-22 at 14 38 37" src="https://github.com/pocketbase/pocketbase/assets/729447/c06b2882-2e17-497c-823f-9ec6cefd88ec">

**Expected**: It would be good if the error was displayed at the row level so other data is not affected and even better if the invalid data can be edited or deleted to fix the inconsistency.

**Steps to reproduce**:
1. Create a table `foo` and a table `bar` with a relation field `foo_id` linking to `foo`.
2. Add this JS hook:
```javascript

onRecordAfterCreateRequest((e) => {
	const collection = $app.dao().findCollectionByNameOrId("bar")
	const record = new Record(collection, {
	    "foo_id": "somebadid",
	})
	
	$app.dao().saveRecord(record)
}, "foo")
```
3. Insert a row into `foo` and go to table `bar` in the admin

**Comment by khrome83** [comment]
I was about to post something similar, but I ran into it with a slightly different issue. 

I was using raw sql to insert catalog entries from a dataset. I had, in go, set the `id` and `parent_id` to int (they are numbers) instead of strings. 

This caused me to insert `0` into `parent_id` when `parent_id` was empty. `0` was not a valid `id` from the dataset so it did not load and would spin and have the same behavior without any feedback in Admin GUI. 

I had to delete the DB and let it be rebuild with migrations to resolve the issue. I could of also dropped the data, but this was the easier path as I only have this programatic inserted data so far. 

The table structure was this -

`id`, `name`, `parent_id`

`parent_id` was allowed to be NULL, and would point as a relation to the same table, and the `id` column. 

Once I converted the datatypes in go to `string` then this was resolved and worked as intended. The zero bug was my fault. But the lack of error messaging and no recovery path within the UI.

Curious what the solution is here. While the workaround is managing the SQLite database externally, that can be really hard for people, especially if this was on a server. It's possible an endpoint could solve this without causing issue, but I didn't try that. 

I think the happy path would be one of the following - 

## Always force a fix - 
1. Give a error message about bad relations. 
2. Ask if user wants to NULL the bad relations?
3. Perform NULL action where relation can't be found

I see this version as preserving the function in Pocketbase, but at the expense of user created data.

## Work, but show broken relationship - 
1. Show table, and render what is possible.
2. On fields where relation is broken, show a red icon about the broken relationship and desired ID (presentable would not be displayed, since no match).
3. Let user self correct. 
5. Ensure any queries don't break through hooks or API because of this broken state.
6. OPTIONAL - Would be nice to have a bulk action to resolve bad relations by NULL them out, or asking for a replacement. More of a nice to have in this scenerio.

## Prevent bad relations in the first place - 
1. With hooks, query builder, or raw sql, prevent a bad relation.

I am not a fan of the last one, because you won't catch all bad actions. For example, a non cascade delete can cause a bad relationship. I don't think preventing it is fool proof and will end up doing one of the other, or something else.

**Comment by ganigeorgiev** [comment]
The JS error should be fixed in [v0.18.7 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.18.7).

The invalid value would be displayed as it is in the records listing, but there will be a tooltip next to the field label in the update form with the missing/invalid ids.

I've added a note to also consider enforcing integrity checks even when saving without data validations, but I'll think a little more on it after v0.19+ as it could cause some unintended side effects when running custom data migrations.
